python/ray/data/datasource/file_based_datasource.py

```diff
@@ -215,7 +215,9 @@ def read_files(
 
                 with _open_file_with_retry(
                     read_path,
-                    lambda: open_input_source(fs, read_path, **open_stream_args),
+                    lambda read_path=read_path: open_input_source(
+                        fs, read_path, **open_stream_args
+                    ),
```

Comment:
Why this change?

Reply:
fix B023 violation when  I've run scripts/format.sh